lib/posix/pthread.c

```diff
@@ -432,80 +457,71 @@ int pthread_create(pthread_t *th, const pthread_attr_t *_attr, void *(*threadrou
 		   void *arg)
 {
 	int err;
-	k_spinlock_key_t key;
 	pthread_barrier_t barrier;
 	struct posix_thread *t = NULL;
-	struct posix_thread_attr attr_storage;
-	struct posix_thread_attr *attr = (struct posix_thread_attr *)_attr;
 
-	if (!pthread_attr_is_valid(attr)) {
+	if (!pthread_attr_is_valid((struct posix_thread_attr *)_attr)) {
```

Comment:
I find it a bit misleading that `pthread_attr_is_valid` is used to check the `posix_thread_attr` instead of `pthread_attr_t`, but this is out of the scope of this PR/commit

Reply:
Need a respin anyway, so I'll make that change.

Reply:
Oh wait - no, that's still completely valid. Is it misleading because we have an internal representation of `pthread_attr_t` (i.e. `struct posix_thread_attr`)? It should be quite safe.

Reply:
> It should be quite safe.

Yeah, it's just the naming, on first glance I thought that the `pthread_attr_is_valid()` should be accepting `pthread_attr_t` as arg, but it is actually expecting `posix_thread_attr`